src/reorderable/ReorderableListComponent.coffee

```diff
@@ -25,6 +25,11 @@ class ReorderableListComponent extends React.Component
       dropItem: null
     }
 
+  componentWillReceiveProps: (nextProps) ->
+    if nextProps.items.length != @state.initialOrder.length
```

Comment:
I'm afraid this won't work in all cases. If the items are reordered outside of the component, this will not refresh the state.